detect.go

```diff
@@ -117,36 +115,36 @@ func Detect(detect DetectFunc, options ...Option) {
 	}
 
 	ctx.Platform.Path = config.arguments[1]
-	if logger.IsDebugEnabled() {
-		logger.Debug(PlatformFormatter(ctx.Platform))
+	if config.logger.IsDebugEnabled() {
```

Comment:
I am not sure why the existing code has double the checks for debug enablement.

config.logger.Debug should be able to handle this on its own without having a guarding if condition. I would vote that we remove this and just call config.logger.Debug (which can internally decide whether or not to log this based on whether debug is enabled).

I would also vote we get rid of `IsDebugEnabled` from the public logger interface.

Reply:
Classically log APIs have an `IsDebugEnabled` style method so that you can avoid debug logging when it's expensive. Not so much the performance of the log operation itself, but if you want to log some output that is expensive to generate. Perhaps for debugging, I want to log all of the files in the application directory, this could be a lot, and walking the file system isn't super speedy. That would be expensive to generate, so I don't want to incur that cost unless debug is actually enabled. Being able to check `IsDebugEnabled` allows me to skip even generating that data. Without it, I'd end up generating the data, calling `Debug` and throwing it away.

Also, less important because it's an implementation detail, but the prior implementation would segfault if you call `logger.Debug` but do not actually have debug logging enabled. Because the call to `logger.Debug` just writes directly to the "debug" writer, and that writer is nil if you don't have debug enabled. I will check and see if I fixed this, I don't think so, as I'm just remembering this fun detail.

I'll fix it so that you don't have to call `IsDebugEnabled` to not crash, but I do think we should keep `IsDebugEnabled` as it does provide an author some additional flexibility. 

The individual places `IsDebugEnabled` is being used could be evaluated though and cleaned up if the operation it's guarding isn't actually expensive to calculate. That would make for cleaner code.

Reply:
But that still begs the question if that is something that should be a part of the logger interface or just a general flag that is exposed as a part of the build context?

Reply:
Thinking about it again, I guess if we are debugging things, the primary way to interact with the user is through logs. So I agree that it is fine to keep it attached to the logging interface.

Reply:
The flag to turn it on/off is `BP_LOG_LEVEL`, setting it to `debug`, so that is tied directly to logging. 

Possibly we want to discuss the name of this flag? We do still support `BP_DEBUG` as well. That is the older flag though and I was planning to ask if we could remove it in v2 and just go with `BP_LOG_LEVEL`. Maybe we double down on `BP_DEBUG` and get rid of `BP_LOG_LEVEL` so it's not tied directly to logging? We also only have one log level, so it's really only on or off. `BP_DEBUG` fits better in that way too.

Regardless, I still think it makes sense to keep that state on the logger. I can't think of a case where I'd be doing something for the purpose of debugging and not have it generate some amount of logging. If I'm calling `logger.Debug` then it is convenient to be able to check the same logger if debug is enabled, rather than look somewhere else. I may be in a method that doesn't have access to the other state, like context, but I'll have the logger to be able to log stuff.